utils/docker/publish_docker.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+set -e
+set +x
+
+trap "cd $(pwd -P)" EXIT
+cd "$(dirname "$0")"
+
+MCR_IMAGE_NAME="playwright/java"
+# GITHUB_REF has a form of `refs/tags/v1.3.0`.
+# TAG_NAME would be `v1.3.0`
+TAG_NAME=${GITHUB_REF#refs/tags/}
+PW_VERSION="${TAG_NAME#v}"
+
+RELEASE_CHANNEL="$1"
+if [[ "${RELEASE_CHANNEL}" == "stable" ]]; then
+  if [[ ! "${PW_VERSION}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]];
+    echo "ERROR: cannot publish stable docker with Playwright version '${PW_VERSION}'"
+    exit 1
+  fi
+elif [[ "${RELEASE_CHANNEL}" == "canary" ]]; then
+
+else
+  echo "ERROR: unknown release channel - ${RELEASE_CHANNEL}"
+  echo "Must be either 'stable' or 'canary'"
+  exit 1
+fi
+
+if [[ -z "${GITHUB_SHA}" ]]; then
+  echo "ERROR: GITHUB_SHA env variable must be specified"
+  exit 1
+fi
+
+BIONIC_TAGS=(
+  "next-bionic"
+  "v${PW_VERSION}-bionic"
+)
+if [[ "$RELEASE_CHANNEL" == "stable" ]]; then
+  BIONIC_TAGS+=("bionic")
+fi
+
+FOCAL_TAGS=(
+  "next"
+  "sha-${GITHUB_SHA}"
+  "next-focal"
+)
+
+if [[ "$RELEASE_CHANNEL" == "stable" ]]; then
+  FOCAL_TAGS+=("latest")
+  FOCAL_TAGS+=("focal")
+  FOCAL_TAGS+=("v${PW_VERSION}-focal")
+  FOCAL_TAGS+=("v${PW_VERSION}")
+fi
+
+tag_and_push() {
+  local source="$1"
+  local target="$2"
+  echo "-- tagging: $target"
+  docker tag $source $target
+  docker push $target
+}
+
+publish_docker_images_with_arch_suffix() {
+  local FLAVOR="$1"
+  local TAGS=()
+  if [[ "$FLAVOR" == "bionic" ]]; then
+    TAGS=("${BIONIC_TAGS[@]}")
+  elif [[ "$FLAVOR" == "focal" ]]; then
+    TAGS=("${FOCAL_TAGS[@]}")
+  else
+    echo "ERROR: unknown flavor - $FLAVOR. Must be either 'bionic' or 'focal'"
+    exit 1
+  fi
+  local ARCH="$2"
+  if [[ "$ARCH" != "amd64" && "$ARCH" != "arm64" ]]; then
+    echo "ERROR: unknown arch - $ARCH. Must be either 'amd64' or 'arm64'"
+    exit 1
+  fi
+  # Prune docker images to avoid platform conflicts
+  docker system prune -fa
+  ./build.sh "--${ARCH}" "${FLAVOR}" "${MCR_IMAGE_NAME}:localbuild"
+
+  for ((i = 0; i < ${#TAGS[@]}; i++)) do
+    local TAG="${TAGS[$i]}"
+    tag_and_push "${MCR_IMAGE_NAME}:localbuild" "playwright.azurecr.io/public/${MCR_IMAGE_NAME}:${TAG}-${ARCH}"
+  done
+}
+
+publish_docker_manifest () {
+  local FLAVOR="$1"
+  local TAGS=()
+  if [[ "$FLAVOR" == "bionic" ]]; then
```

Comment:
no bionic for java

Reply:
Lets keep it to have a smaller diff in the future, its not used anyway (the bottom defines how the functions are called, and we call them only with focal)